### ISSUE TITLE
Model analyzer profile fails for multi input python BE models with different types or different shape dimensions

### ISSUE THREAD

**Issue by HarmitMinhas96** [triton-inference-server/model_analyzer#444]
# Description
Running model analyzer to profile a model fails if the model is a multi-input model with either different data_types (TYPE_UINT8 and TYPE_STRING) and/or different dimensions (3D and 1D). The latest model analyzer, version 1.16, will report: `model_analyzer.model_analyzer_exceptions.TritonModelAnalyzerException: The model test_model was not found in the loaded checkpoint`. An older version, 1.6.0, will just hang indefinitely til it hits the timeout.

It works fine if the multi input model has inputs of the same data_types and both inputs are 1D.

I am able to use this model in the triton server.

# Model analyzer command
```model-analyzer -v profile --model-repository=/tmp/modeldeploy.5uy6kdmn --profile-models=test_model --triton-launch-mode=local --checkpoint-directory=/tmp/tmppd0fpddv --output-model-repository=/tmp/tmppd0fpddv --run-config-search-disable --override-output-model-repository --perf-analyzer-timeout=3600```

# Triton Model Config Example:
```
name: "test_model"
max_batch_size: 1
input {
  name: "input_1"
  data_type: TYPE_UINT8
  dims: 512
  dims: 512
  dims: 3
}
input {
  name: "input_2"
  data_type: TYPE_STRING
  dims: 5
}
output {
  name: "output_1"
  data_type: TYPE_FP32
  dims: -1
}
output {
  name: "output_2"
  data_type: TYPE_STRING
  dims: -1
}
instance_group {
  count: 1
  kind: KIND_GPU
}
backend: "python"
```
# Triton Python BE Model Minimal Example:
```
import os

import numpy as np

import triton_python_backend_utils as pb_utils 


class TritonPythonModel:

  def initialize(self, args):
    pass

  def execute(self, requests):

    responses = []

    for _ in requests:
      out_tensor_1 = pb_utils.Tensor("output_1", np.asarray([[.7]]).astype('float32'))
      out_tensor_2 = pb_utils.Tensor("output_2", np.asarray([['test']]).astype('object'))

      inference_response = pb_utils.InferenceResponse(
          output_tensors=[out_tensor_1, out_tensor_2])
      responses.append(inference_response)

    return responses
```

**Comment by tgerdesnv** [comment]
Hi @HarmitMinhas96, thanks for reporting this.

I tried to reproduce, but ran into issues. I had to change the input type from TYPE_STRING to TYPE_FP32. After that, everything worked fine on 22.05 (1.16).  It also worked fine on (not yet released) 22.06 as well as our main branch.

I suspect that you have an old checkpoint in the checkpoint directory. The checkpoint format isn't guaranteed to work across versions, and we did make some breaking changes a few releases ago.  Can you make sure your checkpoint directory is empty and try again?

**Comment by HarmitMinhas96** [comment]
Hello,

It seems it does work if :
- my inputs are TYPE_UINT8 and TYPE_FP32 with different dimensions
- my first input and second input are both TYPE_STRING with different dimensions
but I need the combination from the specified config.pbtxt, TYPE_UINT8 + TYPE_STRING. I'm not sure why it didn't work for me previously when I tried similar data types with different dimensions, but it seems that isn't a problem.

Also I updated the model.py I specified in the description slightly, it seems model analyzer does not work with empty outputs when testing.

**Comment by tgerdesnv** [comment]
@HarmitMinhas96 it sounds like the only remaining issue is around having input types of UINT8 + STRING.  This is a bug in perf_analyzer that was recently fixed and will be part of the 22.07 release.  [The fix](https://github.com/triton-inference-server/client/pull/117) is in main if you wanted to build perf_analyzer manually.

As a workaround, you could instead [supply custom input data](https://github.com/triton-inference-server/server/blob/main/docs/perf_analyzer.md#real-input-data):
- Create a json file with data:
```
{
        "data" :
        [
                {
                        "input_1" : [1],
                        "input_2" : ["a","b","c","d","e"]
                }
        ]
}
```

- Put that information in a Model Analyzer config.yaml file:
```
perf_analyzer_flags:
  input-data: /path/to/data.json
```

- Run Model Analyzer and pass the config file:
```
model-analyzer -v profile -f /path/to/config.yaml <all_other_options>
```

**Comment by HarmitMinhas96** [comment]
It works when saving it as a 1D array in the input-data file. I can use this as a workaround until the release.
Thank you!
